app/jobs/fetch_globus_job.rb

```diff
@@ -20,12 +20,9 @@ def perform(work_version)
   end
 
   def filepaths_for(work_version)
-    filepaths = GlobusClient.get_filenames(path: work_version.globus_endpoint, user_id: work_version.work.owner.email)
-    filepaths.map { |filepath| filepath.delete_prefix(globus_prefix(work_version)) }
-  end
-
-  def globus_prefix(work_version)
-    "#{Settings.globus.uploads_directory}#{work_version.globus_endpoint}/"
+    GlobusClient
+      .get_filenames(path: work_version.globus_endpoint, user_id: work_version.work.owner.email)
+      .map { |filepath| filepath.delete_prefix(work_version.globus_endpoint_fullpath) }
```

Comment:
Two changes here:

1. Ditch the local variable since it's already clear that we're dealing with a list of file paths.
2. Replace the `#globus_prefix` method in the job with a call to `#globus_endpoint_fullpath` on the `WorkVersion` instance